canvas/src/layer/layer_2d.rs

```diff
@@ -3,32 +3,169 @@ use crate::{
 };
 pub use glyph_brush::{GlyphCruncher, HorizontalAlign, Layout, VerticalAlign};
 pub use lyon::{
-  path::{builder::PathBuilder, traits::PathIterator, Path, Winding},
+  geom::Arc,
+  math::{point, rect},
+  path::{builder::PathBuilder, path::Builder, traits::PathIterator, EndpointId, Path, Winding},
   tessellation::*,
 };
+
 use std::{
   cmp::PartialEq,
   ops::{Deref, DerefMut},
 };
 
+pub type Vector = euclid::default::Vector2D<f32>;
```

Comment:
Should assign a unit `LogicUnit` to `Vector` and `Angle` type,  define in `lib.rs` is better ?